capabilities/lib/kubernetes-api.ts

```diff
@@ -1,194 +1,145 @@
-import {
-  AppsV1Api,
-  CoreV1Api,
-  CustomObjectsApi,
-  KubeConfig,
-  V1Ingress,
-  NetworkingV1Api,
-} from "@kubernetes/client-node";
-
-import {
-  VirtualService,
-  Gateway,
-} from "@kubernetes-models/istio/networking.istio.io/v1beta1";
-
-type K8sModel = {
-  apiVersion: string;
-  kind: string;
-};
-
-function getGroupVersionPlural(model: K8sModel) {
-  const [group, version] = model.apiVersion.split("/");
-  const plural = model.kind.toLocaleLowerCase() + "s";
-  return { group, version, plural };
-}
+import { GenericClass } from "kubernetes-fluent-client";
+import crypto from "crypto";
+import { Log, kind, K8s } from "pepr";
 
 export class K8sAPI {
-  k8sApi: CoreV1Api;
-  k8sAppsV1Api: AppsV1Api;
-  k8sCustomObjectsApi: CustomObjectsApi;
-  networkingV1Api: NetworkingV1Api;
-
-  constructor() {
-    const kc = new KubeConfig();
-    kc.loadFromDefault();
-    this.k8sApi = kc.makeApiClient(CoreV1Api);
-    this.k8sAppsV1Api = kc.makeApiClient(AppsV1Api);
-    this.k8sCustomObjectsApi = kc.makeApiClient(CustomObjectsApi);
-    this.networkingV1Api = kc.makeApiClient(NetworkingV1Api);
+  static async labelNamespaceForIstio(namespace: string) {
+    return await K8s(kind.Namespace).Apply({
+      metadata: { name: namespace, labels: { "istio-injection": "enabled" } },
+    });
   }
 
-  async upsertVirtualService(virtualService: VirtualService) {
-    const { group, version, plural } = getGroupVersionPlural(VirtualService);
+  static async checksumApp(
```

Comment:
i don't understand the reason for this? Can you give a description in the PR to make it more obvious?

Reply:
adding a checksum to an annotation will restart the pods per the strategy defined by the application. This code will support statefulsets, daemonsets and deployments. So if any of the pods in those apps do not have a sidecar, this method will push that in. we've used this exact method with authservice to let the deployment know it's underlying secrets changed.